app/src/main/java/com/example/groceryshop/activities/fragment/FrmLogin.java

```diff
@@ -60,15 +80,90 @@ protected void loadControlsAndResize(View view) {
         icKey.getLayoutParams().height = activity.getSizeWithScale(11);
         icKey.getLayoutParams().width = activity.getSizeWithScale(11);
 
+        edtEmailLogin = view.findViewById(R.id.edtEmailLogin);
+        edtPassLogin = view.findViewById(R.id.edtPassLogin);
+        ckbRemember = view.findViewById(R.id.ckbRemember);
+
         view.findViewById(R.id.tvSignUp).setOnClickListener(this);
+        view.findViewById(R.id.tvForgotPass).setOnClickListener(this);
+        btnLogin.setOnClickListener(this);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        sharedPreferences = Preferences.getPreferences().newSharedPreferences(getContext());
+        boolean check = sharedPreferences.getBoolean(PrefConstants.PREF_CHECK_SAVE_USER, false);
+        if (check) {
+            if (sharedPreferences != null) {
+                edtEmailLogin.setText(sharedPreferences.getString(PrefConstants.PREF_EMAIL_SAVE_USER, ""));
+                edtPassLogin.setText(sharedPreferences.getString(PrefConstants.PREF_PASSWORD_SAVE_USER, ""));
+                ckbRemember.setChecked(check);
+            }
+        } else {
+            edtEmailLogin.setText("");
+            edtPassLogin.setText("");
+            ckbRemember.setChecked(false);
+        }
+
     }
 
     @Override
     public void onClick(View v) {
-        switch (v.getId()){
+        switch (v.getId()) {
             case R.id.tvSignUp:
                 activity.showFrmSignUp();
                 break;
+            case R.id.btnLogin:
+                loginUser();
+                break;
+            case R.id.tvForgotPass:
+                activity.showFrmForgotPassword();
+                break;
+        }
+    }
+
+    //TODO Dummy API
+    private ListenerAPI listenerAPI = new ListenerAPI() {
+        @Override
+        public void onStarts() {
+            activity.showDialogLoading();
         }
+
+        @Override
+        public void onResult(boolean isSuccess) {
+            activity.dismissDialog();
+
+            if (isSuccess) {
+                Log.e(TAG, "onResult: " + isSuccess );
+                UserEntity userEntity = DatabaseHelper.getDatabaseHelper(getContext()).Login(new UserEntity(email, pass));
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.